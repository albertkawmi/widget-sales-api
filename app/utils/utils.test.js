/* eslint-env jest */
const { loggingService, get404message } = require('./')

const consoleLog = jest.spyOn(console, 'log')
  .mockImplementation(() => {})

// Mocks
const req = {
  method: 'GET',
  originalUrl: '/clients',
  hostname: 'localhost',
  ip: '1.2.3.4'
}
const res = {}
const next = () => {}

test('loggingService', () => {
  process.env.NODE_ENV = 'test'
  loggingService(req, res, next)
  expect(consoleLog).not.toHaveBeenCalled()

  process.env.NODE_ENV = 'production'
  loggingService(req, res, next)
  expect(consoleLog).toHaveBeenCalledWith(
    expect.stringMatching(/GET \/clients localhost 1\.2\.3\.4/)
  )
})

test('get404message', () => {
  expect(get404message('/not/a/page'))
    .toMatchSnapshot()
})
