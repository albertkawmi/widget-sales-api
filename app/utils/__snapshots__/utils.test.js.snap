// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get404message 1`] = `"Path /not/a/page does not exist. Please see https://github.com/albertkawmi/widget-sales-api for more information."`;
